
 
cd /DjangoGather/WebServer/服务器后端/福平台
 
django-admin startapp system
django-admin startapp demo
django-admin startapp generator



# 细节
非当前目录下 创建app 需要修改 apps -> name = '福平台.system' 修改路径 ...


# 修改日志
# 数据库迁移生效
python manage.py makemigrations
python manage.py migrate

python manage.py makemigrations 福平台.generator



root@4b64bfc5a9b1:/DjangoGather/WebServer/服务器后端# python manage.py makemigrations
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/django/db/models/utils.py", line 15, in make_model_tuple
    app_label, model_name = model.split(".")
    ^^^^^^^^^^^^^^^^^^^^^
ValueError: too many values to unpack (expected 2)

During handling of the above exception, another exception occurred:

Traceback (mos

       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/django/db/models/fields/related.py", line 86, in <genexpr>
    model_keys = (make_model_tuple(m) for m in models)
                  ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/django/db/models/utils.py", line 22, in make_model_tuple
    raise ValueError(
ValueError: Invalid model reference '福平台.system.Users'. String model references must be of the form 'app_label.ModelName'.
root@4b64bfc5a9b1:/DjangoGather/WebServer/服务器后端#
root@4b64bfc5a9b1:/DjangoGather/WebServer/服务器后端#


INSTALLED_APPS = [
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',

    '福平台.system',
    '福平台.demo',
    '福平台.generator',

]


# # 变量设置 
AUTH_USER_MODEL = '福平台.system.Users'


对于我这种多层文件夹下管理的app 来说应该怎么做才可以用这样的结构使用 用户模型
